stop removal of spaces that causes display changes

tisdall opened the pull request:
I can't believe how much time I spent on this...  😄   

This PR fixes some additional cases of removal of spaces that changes the display.  Most of the cases are poorly formed HTML, but unfortunately we're not always in control over that.  In some edge cases it's leaving a single space that doesn't affect display in the browsers I tested, but I think this keeps things simple and errs on the side of leaving a space that _may_ affect display.  I looked further into #121 and #21 and neither of them are actually bugs in the latest version.

### Examples
(I'm going to exclude the `<html>` wrappers in the output for simplicity)

#### Example 1:
Stop removal of needed space for proper display.

Un-minified: `<p> <i>first </i>second </p>` which displays as: <i>first</i> second
django-htmlmi 0.11.0: `<p><i>first</i>second</p>` which displays as: <i>first</i>second
this PR: `<p><i>first </i>second</p>` which displays as: <i>first</i> second

#### Example 2:
Stop the removal of a space that has styling applied to it (hard to duplicate in github, but you can see the space is underlined between the words)

Un-minified: `<p><u>first </u> <s> second</s></p>` which displays as: <ins>first </ins> <s> second</s>
django-htmlmi 0.11.0: `<p><u>first</u> <s>second</s></p>` which displays as: <ins>first</ins> <s>second</s>
this PR: `<p><u>first </u> <s> second</s></p>` which displays as: <ins>first </ins> <s> second</s>

This one is weird as the order dictates what styling that space in the middle receives.  If you switch the order then the space in the middle gets a strike-through styling, but I have no idea if that's consistent across all browser renderers.

andrewsmedina commented on htmlmin/minify.py:
```diff
@@ -175,23 +175,19 @@ def is_cond_comment(soup):
+    def is_text_tag(soup):
+        return soup is not None and soup.name in TEXT_FLOW
+
+    if is_text_tag(soup.parent):
+        # NavigableString is within a text tag element
+        return (True, True)
```
This function always returns the same value. Why this is necessary?

tisdall replied:
Essentially, this is an optimization.  Prior to this, I had something like this:
```python
prev_flow = (
    is_text_tag(soup.previous_sibling)
    or is_text_tag(soup.parent)
)

next_flow = (
    is_text_tag(soup.next_sibling)
    or is_text_tag(soup.parent)
)
```

Then I noticed that I was essentially running `is_text_tag(soup.parent)` twice and that when `True` it forces both `prev_flow` and `next_flow` to `True`.

I'm open to suggestions on a code comment to make it more understandable.

andrewsmedina commented on htmlmin/minify.py:
```diff
@@ -175,23 +175,19 @@ def is_cond_comment(soup):
-
-    return (prev_flow, next_flow)
+    def is_text_tag(soup):
+        return soup is not None and soup.name in TEXT_FLOW
```
This code can be simplified by
```suggestion
        return soup and soup.name in TEXT_FLOW
```

tisdall replied:
True.  I had kept it that way as the previous code was like that, but it makes sense to change it.

andrewsmedina reviewed (changes requested):
Thanks for contributing. There are few things that can be improved and there is a test broken that should be fixed.

tisdall commented:
I must have accidentally hit space while having the test_minify.py open or something as I had run the tests locally.  I've updated that one test and made the suggested change and updated the PR.

andrewsmedina reviewed (approved):
LGTM
